index.js

```diff
@@ -1,358 +1,451 @@
-
-    // Current relative humidity
-    device
-      .getCharacteristic(Characteristic.CurrentRelativeHumidity)
-      .on('get', this.getCurrentRelativeHumidity.bind(this))
-
-    // Target relative humidity
-    device
-      .getCharacteristic(Characteristic.RelativeHumidityHumidifierThreshold)
-      .setProps({
-        minValue: 30,
-        maxValue: 80,
-        minStep: 10,
-        validValues: [30, 40, 50, 60, 70, 80]
-      })
-      .on('get', this.getTargetRelativeHumidity.bind(this))
-      .on('set', this.setTargetRelativeHumidity.bind(this))
-
-    // Current water level (remaining water level)
-    // Note: this characteristic works only for Smartmi Evaporative Humidifier
-    isModel2 && device
-      .getCharacteristic(Characteristic.WaterLevel)
-      .on('get', this.getWaterLevel.bind(this))
-
-    // Rotation speed
-    device
-      .getCharacteristic(Characteristic.RotationSpeed)
-      .setProps({
-        minValue: 0,
-        maxValue: isModel2 ? 4 : 3,
-        minStep: 1
-      })
-      .on('get', this.getRotationSpeed.bind(this))
-      .on('set', this.setRotationSpeed.bind(this))
-
-    // Child lock
-    // Note: this characteristic works only for Smartmi Evaporative Humidifier
-    isModel2 && device
-      .addCharacteristic(Characteristic.LockPhysicalControls)
-      .on('get', this.getLockPhysicalControls.bind(this))
-      .on('set', this.setLockPhysicalControls.bind(this))
-
-    // Drying mode
-    // Note: this characteristic works only for Smartmi Evaporative Humidifier
-    // TODO: maybe here we need to use something else instead of SwingMode, but this is the closest Characteristic type
-    isModel2 && device
-      .addCharacteristic(Characteristic.SwingMode)
-      .on('get', this.getDryingMode.bind(this))
-      .on('set', this.setDryingMode.bind(this))
-
-    // Temperature sensor
-    if (options.showTemperature) {
-      let temperature = new Service.TemperatureSensor(options.nameTemperature),
-        handler = options.model === 'cb1' ? this.getCurrentTemperatureCB1 : this.getCurrentTemperature
-
-      temperature
-        .getCharacteristic(Characteristic.CurrentTemperature)
-        .on('get', handler.bind(this))
-
-      this.services.push(temperature)
-    }
+    constructor(log, config) {
+        if (!config.ip) throw new Error('Your must provide IP address of the Humidifier');
+        if (!config.token) throw new Error('Your must provide token of the Humidifier');
+
+        let options = { ...defaults, ...config };
+        this.infoService = new Service.AccessoryInformation();
+        this.humidifierService = new Service.HumidifierDehumidifier(options.name);
+
+        this.log = log;
+        this.ip = config.ip;
+        this.token = config.token;
+        this.services = [this.humidifierService, this.infoService];
+        this.humidifier = SUPPORTED_HUMIDIFIERS.find(element => element.version === config.model);
```

Comment:
```suggestion
        this.humidifier = SUPPORTED_HUMIDIFIERS.find(element => element.version === options.model);
```

Reply:
Otherwise when there is no `model` in config.json default `v1` won't be applied

Reply:
Fixed